http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/NettySdkHttpClientFactory.java

```diff
@@ -136,15 +163,40 @@ public static Builder builder() {
         Builder maxConnectionsPerEndpoint(Integer maxConnectionsPerEndpoint);
 
         /**
-         * The amount of time to wait for data to be transferred over an established, open connection before the connection is
-         * timed out.
+         * The amount of time to wait for a read on a socket before an exception is thrown.
+         * <br/>
+         * <strong>note: minimum supported granularity is seconds, if {@link Duration} cannot be converted
+         * to a whole second an exception will be thrown</strong>
+         *
+         * @param timeout timeout duration
+         * @return this builder for method chaining.
          */
-        Builder socketTimeout(Duration socketTimeout);
+        Builder readTimeout(Duration timeout);
+
+        /**
+         * The amount of time to wait for a write on a socket before an exception is thrown.
+         * <br/>
+         * <strong>note: minimum supported granularity is seconds, if {@link Duration} cannot be converted
+         * to a whole second an exception will be thrown</strong>
+         *
+         * @param timeout timeout duration
+         * @return this builder for method chaining.
+         */
+        Builder writeTimeout(Duration timeout);
 
         /**
          * The amount of time to wait when initially establishing a connection before giving up and timing out.
+         * @param timeout the timeout duration
+         * @return this builder for method chaining.
+         */
+        Builder connectionTimeout(Duration timeout);
```

Comment:
Not sure if it makes sense for `connectionTimeout` and `connectionAcquisitionTimeout` to be different. In our apache configuration we use the same value for both "new connection" (Apache calls this `connectionTimeout`) and "get connection from pool" (Apache calls this `connectionRequestTimeout`) see here : https://github.com/aws/aws-sdk-java-v2/blob/master/http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/internal/impl/ApacheHttpRequestFactory.java#L68:L69

In any case the default for `connectionAcquisitionTimeout` (if not set) should probably be to use `connectionTimeout`